### PR TITLE
use ByteArraySerializer for producer

### PR DESCRIPTION
**Pull request by adyach**
Nakadi needs to use byte serializers in order to work with binary data, the change is tech refactoring as precondition for binary formats in Nakadi

### CODE REVIEW

**Comment by adyach** [comment]
👍

File: core-common/src/test/java/org/zalando/nakadi/domain/EventOwnerHeaderTest.java [hunk 1]
```diff
@@ -14,7 +16,9 @@
     @Test
     public void testEventOwnerHeaderSerialization() {
         final EventOwnerHeader eventOwnerHeader = new EventOwnerHeader(NAME, VALUE);
-        final ProducerRecord<String, String> record = new ProducerRecord<>("topic", "value");
+        final ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(
+                "topic",
```
Comment (a1exsh) [inline core-common/src/test/java/org/zalando/nakadi/domain/EventOwnerHeaderTest.java:20]: How does it work here with "topic" not converted to `byte[]`?

Comment (adyach) [reply core-common/src/test/java/org/zalando/nakadi/domain/EventOwnerHeaderTest.java:20]: conversion to bytes only for key / value, that should be non default producer constructor where key is null by default

Comment (adyach) [reply core-common/src/test/java/org/zalando/nakadi/domain/EventOwnerHeaderTest.java:20]: yes, that is exactly the case 
```
    public ProducerRecord(String topic, V value) {
        this(topic, null, null, null, value, null);
    }
```

Comment (a1exsh) [reply core-common/src/test/java/org/zalando/nakadi/domain/EventOwnerHeaderTest.java:20]: ah, I see.  the first argument is not key, but indeed a topic, which is expected to be of type `String`, regardless of the key and value types: https://kafka.apache.org/23/javadoc/org/apache/kafka/clients/producer/ProducerRecord.html#ProducerRecord-java.lang.String-V-
```
public ProducerRecord(String topic,
                      V value)
Create a record with no key
```

**Comment by a1exsh** [comment]
:+1:

**Pull request merged**